frontend/model/contracts/group.js

```diff
@@ -1587,59 +1576,6 @@ sbp('chelonia/defineContract', {
       ).catch(e => {
         console.error(`[gi.contracts/group/_cleanup] Error removing entries for archive for ${contractID}`, e)
       })
-
-      // If we're only cleaning up after receiving new keys, we're not actually
-      // leaving the group and we don't need to do any of the actions related
-      // to leaving
-      if (resync) return
-
-      // grab the groupID of any group that we're a part of
-      if (!rootState.currentGroupId || rootState.currentGroupId === contractID) {
-        const groupIdToSwitch = Object.keys(contracts)
-          .filter(cID =>
-            contracts[cID].type === 'gi.contracts/group' &&
-              cID !== contractID
-          ).sort(cID =>
-          // prefer successfully joined groups
-            rootState[cID]?.profiles?.[identityContractID] ? -1 : 1
-          )[0] || null
-        sbp('state/vuex/commit', 'setCurrentChatRoomId', {})
-        sbp('state/vuex/commit', 'setCurrentGroupId', groupIdToSwitch)
-      }
-
-      // Remove last logged in information
-      Vue.delete(rootState.lastLoggedIn, contractID)
```

Comment:
Is this all now done somewhere else? Or is it no longer necessary because...?

Reply:
https://github.com/okTurtles/group-income/pull/1955/files/ccdfb7599dc4240066204d943cb5ff044aebd3d0#diff-eaaa3e4a455bba1ad6131a29aec3a915aa1f656241bc956c556e341812bbd42eR312-R327

Reply:
This has been moved to the identity contract, which keeps track of the groups one is member of.